Sources/SwiftParser/Lexer/Cursor.swift

```diff
@@ -335,6 +337,8 @@ extension Lexer.Cursor {
       flags.insert(.isAtStartOfLine)
     }
 
+    self.previousTokenKind = result.tokenKind.base
```

Comment:
There's `Cursor.backUp` where `previousTokenKind` wouldn't be correct any more. We also don't set the state in there, but I suppose the assumption there is that we're always in `normal` when splitting tokens.

Reply:
`Cursor.backUp` is still correct because the previous token still has the same kind if we `resetForSplit`, which is the only caller of this function (and should remain to be the only caller). 

If we have just lexed `<..` as a binary operator, `previousToken` is `binaryOperator`. If we now back up by two characters to consume only `<`, we are now placed at the first `.` but the previous token continues to be of kind `binaryOperator`. So no change of `previousTokenKind`  is necessary here.